lib/Driver/FrontendUtil.cpp

```diff
@@ -57,6 +58,25 @@ bool swift::driver::getSingleFrontendInvocationFromDriverArguments(
   if (Diags.hadAnyError())
     return true;
 
+  if (ForceNoOutputs) {
+    // Clear existing output modes.
+    ArgList->eraseArg(options::OPT_modes_Group);
+    // Disable other output options that conflict with -typecheck.
+    ArgList->eraseArg(options::OPT_emit_module);
+    ArgList->eraseArg(options::OPT_emit_module_path);
+    ArgList->eraseArg(options::OPT_emit_module_source_info_path);
+    ArgList->eraseArg(options::OPT_emit_module_interface);
+    ArgList->eraseArg(options::OPT_emit_module_interface_path);
+    ArgList->eraseArg(options::OPT_emit_objc_header);
+    ArgList->eraseArg(options::OPT_emit_objc_header_path);
```

Comment:
More you might want to disable (plus corresponding `_path`s):

* OPT_emit_dependencies
* OPT_emit_loaded_module_trace
* OPT_emit_private_module_interface_path
* OPT_emit_tbd

There's also OPT_serialize_diagnostics, but I'm guessing you may not want to touch that one.

-----

I worry that this list won't be kept up to date as people add new entries to Options.td. Would it make sense to make a new OptionFlag for "options that cause supplementary outputs to be emitted"? Then you could erase everything from that category (perhaps with some exceptions, like -serialize-diagnostics) and people would just need to appropriately annotate entries when they added them to the options table.

Reply:
> There's also OPT_serialize_diagnostics, but I'm guessing you may not want to touch that one.

Why not? We generally shouldn't be serializing diagnostics to a file in sourcekitd.

> Would it make sense to make a new OptionFlag for "options that cause supplementary outputs to be emitted"?

Good idea!